### ISSUE TITLE
Can not download as PDF when contains chinese words

### ISSUE THREAD

**Issue by netroby** [jupyter/notebook#1674]
Can not download as PDF when contains chinese words

**Comment by netroby** [comment]
http://nbviewer.jupyter.org/github/netroby/jupyter-notebook/blob/master/How%20to%20write%20python.ipynb

**Comment by netroby** [comment]
The Error log
```
500 : Internal Server Error
The error was:
nbconvert failed: PDF creating failed, captured latex output:
This is pdfTeX, Version 3.14159265-2.6-1.40.15 (TeX Live 2015/dev/Debian) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./notebook.tex
LaTeX2e <2014/05/01>
Babel <3.9l> and hyphenation patterns for 2 languages loaded.
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size11.clo))
(/usr/share/texlive/texmf-dist/tex/latex/base/fontenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/t1enc.def))
(/usr/share/texlive/texmf-dist/tex/latex/psnfss/palatino.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/graphics.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/pdftex-def/pdftex.def
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ltxcmds.sty))))
(/usr/share/texlive/texmf-dist/tex/latex/caption/caption.sty
(/usr/share/texlive/texmf-dist/tex/latex/caption/caption3.sty))
(/usr/share/texlive/texmf-dist/tex/latex/adjustbox/adjustbox.sty
(/usr/share/texlive/texmf-dist/tex/latex/xkeyval/xkeyval.sty
(/usr/share/texlive/texmf-dist/tex/generic/xkeyval/xkeyval.tex
(/usr/share/texlive/texmf-dist/tex/generic/xkeyval/xkvutils.tex)))
(/usr/share/texlive/texmf-dist/tex/latex/adjustbox/adjcalc.sty)
(/usr/share/texlive/texmf-dist/tex/latex/adjustbox/trimclip.sty
(/usr/share/texlive/texmf-dist/tex/latex/collectbox/collectbox.sty)
(/usr/share/texlive/texmf-dist/tex/latex/adjustbox/tc-pdftex.def))
(/usr/share/texlive/texmf-dist/tex/latex/ifoddpage/ifoddpage.sty)
(/usr/share/texlive/texmf-dist/tex/latex/varwidth/varwidth.sty))
(/usr/share/texlive/texmf-dist/tex/latex/xcolor/xcolor.sty
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/color.cfg))
(/usr/share/texlive/texmf-dist/tex/latex/tools/enumerate.sty)
(/usr/share/texlive/texmf-dist/tex/latex/geometry/geometry.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifpdf.sty)
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifvtex.sty)
(/usr/share/texlive/texmf-dist/tex/generic/ifxetex/ifxetex.sty))
(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amstext.sty
(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsgen.sty))
(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsbsy.sty)
(/usr/share/texlive/texmf-dist/tex/latex/amsmath/amsopn.sty))
(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amssymb.sty
(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/amsfonts.sty))
(/usr/share/texlive/texmf-dist/tex/latex/base/textcomp.sty
(/usr/share/texlive/texmf-dist/tex/latex/base/ts1enc.def))
(/usr/share/texlive/texmf-dist/tex/latex/upquote/upquote.sty)
(/usr/share/texlive/texmf-dist/tex/latex/eurosym/eurosym.sty)
(/usr/share/texlive/texmf-dist/tex/latex/ucs/ucs.sty
(/usr/share/texlive/texmf-dist/tex/latex/ucs/data/uni-global.def))
(/usr/share/texlive/texmf-dist/tex/latex/base/inputenc.sty
(/usr/share/texlive/texmf-dist/tex/latex/ucs/utf8x.def))
(/usr/share/texlive/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty
Style option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix 
<2008/02/07> (tvz))
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grffile.sty
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/kvoptions.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/etexcmds.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/ifluatex.sty))))
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty))
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/auxhook.sty)
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/hyperref.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/url/url.sty))

Package hyperref Message: Driver (autodetected): hpdftex.

(/usr/share/texlive/texmf-dist/tex/latex/hyperref/hpdftex.def
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))
(/usr/share/texlive/texmf-dist/tex/latex/tools/longtable.sty)
(/usr/share/texlive/texmf-dist/tex/latex/booktabs/booktabs.sty)
(/usr/share/texlive/texmf-dist/tex/generic/ulem/ulem.sty)
No file notebook.aux.
(/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texlive/texmf-dist/tex/latex/psnfss/t1ppl.fd)
(/usr/share/texlive/texmf-dist/tex/context/base/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texlive/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty
(/usr/share/texlive/texmf-dist/tex/latex/oberdiek/grfext.sty)
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
(/usr/share/texlive/texmf-dist/tex/latex/caption/ltcaption.sty)
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex
*geometry* verbose mode - [ preamble ] result:
* driver: pdftex
* paper: <default>
* layout: <same size as paper>
* layoutoffset:(h,v)=(0.0pt,0.0pt)
* modes: 
* h-part:(L,W,R)=(72.26999pt, 469.75502pt, 72.26999pt)
* v-part:(T,H,B)=(72.26999pt, 650.43001pt, 72.26999pt)
* \paperwidth=614.295pt
* \paperheight=794.96999pt
* \textwidth=469.75502pt
* \textheight=650.43001pt
* \oddsidemargin=0.0pt
* \evensidemargin=0.0pt
* \topmargin=-37.0pt
* \headheight=12.0pt
* \headsep=25.0pt
* \topskip=11.0pt
* \footskip=30.0pt
* \marginparwidth=59.0pt
* \marginparsep=10.0pt
* \columnsep=10.0pt
* \skip\footins=10.0pt plus 4.0pt minus 2.0pt
* \hoffset=0.0pt
* \voffset=0.0pt
* \mag=1000
* \@twocolumnfalse
* \@twosidefalse
* \@mparswitchfalse
* \@reversemarginfalse
* (1in=72.27pt=25.4mm, 1cm=28.453pt)

(/usr/share/texlive/texmf-dist/tex/latex/ucs/ucsencs.def)
(/usr/share/texlive/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/share/texlive/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))
(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/umsa.fd)
(/usr/share/texlive/texmf-dist/tex/latex/amsfonts/umsb.fd)

LaTeX Warning: No \author given.

(/usr/share/texlive/texmf-dist/tex/latex/ucs/data/uni-79.def)
(/usr/share/texlive/texmf-dist/tex/latex/ucs/data/uninames.dat)
(/usr/share/texlive/texmf-dist/tex/latex/ucs/data/uni-79.def)

! Package ucs Error: Unknown Unicode character 20320 = U+4F60,
(ucs)                possibly declared in uni-79.def.
(ucs)                Type H to see if it is available with options.

See the ucs package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.284 你
         们这些磨人的小妖精啊．非要老夫弄什么中文，弄�...

? 
! Emergency stop.
 ...                                              
                                                  
l.284 你
         们这些磨人的小妖精啊．非要老夫弄什么中文，弄�...

!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on notebook.log.
```

**Comment by takluyver** [comment]
If you have time, can you try installing [nbconvert](https://github.com/jupyter/nbconvert) from master? We've switched from pdflatex to xelatex, which is supposed to handle unicode better.

**Comment by netroby** [comment]
Sorry  i do not know how to install from master branch

**Comment by takluyver** [comment]
If you have git:

```
git clone https://github.com/jupyter/nbconvert.git
cd nbconvert
pip install -e .
```

Or shorter:

```
pip install git+https://github.com/jupyter/nbconvert.git
```

**Comment by gnestor** [comment]
@takluyver I'm trying to test this but I'm getting the following error: `nbconvert failed: pdflatex not found on PATH`. I saw a couple similar issues on nbconvert, etc. I assume pdflatex needs to be installed (e.g. via Homebrew) in order to use "Download as PDF"? Same with xelatex? If so, the nbconvert docs need to reflect this.

**Comment by takluyver** [comment]
Yes, conversion to PDF requires Latex, and yes, we ought to make that clear in the docs.

**Comment by gnestor** [comment]
Related https://github.com/jupyter/nbconvert/issues/281

**Comment by michaelpacer** [comment]
Also, I would not recommend installing pdflatex with homebrew. If possible
please install using https://tug.org/mactex/ ; it will make sure that all
of the necessary packages are present.

I'm not sure if there's an easy way to use mactex from the command line.
I'm sure it's possible, if needed I'll look into it.

**Comment by gnestor** [comment]
@michaelpacer Ya I discovered that TeX binaries are shipped OS X but not included in the PATH in OS X El Capitan (https://github.com/jupyter/nbconvert/issues/406#issue-176941534), so adding the path to the binaries to the PATH fixes the `nbconvert failed: pdflatex not found on PATH` issue. I'm running into a new issue: 

```
nbconvert failed: PDF creating failed, captured latex output:
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./notebook.tex
LaTeX2e <2016/03/31>
Babel <3.9r> and hyphenation patterns for 22 language(s) loaded.
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/base/size11.clo))
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/base/fontenc.sty
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/base/t1enc.def))
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/psnfss/palatino.sty)
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/pdftex-def/pdftex.def
(/usr/local/texlive/2016basic/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/local/texlive/2016basic/texmf-dist/tex/generic/oberdiek/ltxcmds.sty))))
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/caption/caption.sty
(/usr/local/texlive/2016basic/texmf-dist/tex/latex/caption/caption3.sty))

! LaTeX Error: File `adjustbox.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
! Emergency stop.
<read *> 
         
l.42     \usepackage
                    {xcolor} % Allow colors to be defined^^M
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on notebook.log.
```

**Comment by michaelpacer** [comment]
Did you install pdflatex/xelatex via homebrew? I really don't recommend installing them separately from the entire macTeX package. 

That isn't something that ships with OS X, it's a completely separate package that installs basically everything you need for a modern TeX installation (including many of the packages on CTAN). 

If you build your LaTeX system from scratch using macTeX (which you might be able to get from CTAN, I now see: https://www.ctan.org/pkg/mactex) I'd recommend trying that rather than debugging some stuff done by brew.

**Comment by michaelpacer** [comment]
@netroby Did installing from master work for you?

**Comment by michaelpacer** [comment]
@gnestor So i also found this issue related to `adjust_box.sty`, I'm not sure if it's related to your issue, but I figured it'd help to put it on your radar. https://github.com/jupyter/nbconvert/issues/357

**Comment by gnestor** [comment]
@michaelpacer Before installing anything, I received the `nbconvert failed: pdflatex not found on PATH` error. After installing [basictex](https://tug.org/mactex/morepackages.html), I received the `File 'adjustbox.sty' not found` error. After upgrading to regular [mactex](https://tug.org/mactex/mactex-download.html), I was able to download a notebook as PDF. 

@netroby I tried opening your example notebook and downloading as PDF and I received the same error:

```
nbconvert failed: PDF creating failed, captured latex output:
This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./notebook.tex
LaTeX2e <2016/03/31>
Babel <3.9r> and hyphenation patterns for 83 language(s) loaded.
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/article.cls
Document Class: article 2014/09/29 v1.4h Standard LaTeX document class
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/size11.clo))
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/fontenc.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/t1enc.def))
(/usr/local/texlive/2016/texmf-dist/tex/latex/psnfss/palatino.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/local/texlive/2016/texmf-dist/tex/latex/pdftex-def/pdftex.def
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/infwarerr.sty)
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/ltxcmds.sty))))
(/usr/local/texlive/2016/texmf-dist/tex/latex/caption/caption.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/caption/caption3.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/adjustbox/adjustbox.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/xkeyval/xkeyval.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/xkeyval/xkeyval.tex
(/usr/local/texlive/2016/texmf-dist/tex/generic/xkeyval/xkvutils.tex)))
(/usr/local/texlive/2016/texmf-dist/tex/latex/adjustbox/adjcalc.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/adjustbox/trimclip.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/collectbox/collectbox.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/adjustbox/tc-pdftex.def))
(/usr/local/texlive/2016/texmf-dist/tex/latex/ifoddpage/ifoddpage.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/varwidth/varwidth.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/xcolor/xcolor.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/graphics-cfg/color.cfg))
(/usr/local/texlive/2016/texmf-dist/tex/latex/tools/enumerate.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/geometry/geometry.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/ifpdf.sty)
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/ifvtex.sty)
(/usr/local/texlive/2016/texmf-dist/tex/generic/ifxetex/ifxetex.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsmath/amstext.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsmath/amsgen.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsmath/amsbsy.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsmath/amsopn.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsfonts/amssymb.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsfonts/amsfonts.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/textcomp.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/ts1enc.def))
(/usr/local/texlive/2016/texmf-dist/tex/latex/upquote/upquote.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/eurosym/eurosym.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/ucs.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/data/uni-global.def))
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/inputenc.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/utf8x.def))
(/usr/local/texlive/2016/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty
Style option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix 
<2008/02/07> (tvz))
(/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/grffile.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/kvoptions.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/etexcmds.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/ifluatex.sty))))
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/hyperref/hyperref.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/auxhook.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/hyperref/pd1enc.def)
(/usr/local/texlive/2016/texmf-dist/tex/latex/latexconfig/hyperref.cfg)
(/usr/local/texlive/2016/texmf-dist/tex/latex/url/url.sty))

Package hyperref Message: Driver (autodetected): hpdftex.

(/usr/local/texlive/2016/texmf-dist/tex/latex/hyperref/hpdftex.def
(/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/tools/longtable.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/booktabs/booktabs.sty)
(/usr/local/texlive/2016/texmf-dist/tex/generic/ulem/ulem.sty)
No file notebook.aux.
(/usr/local/texlive/2016/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/local/texlive/2016/texmf-dist/tex/latex/psnfss/t1ppl.fd)
(/usr/local/texlive/2016/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty
(/usr/local/texlive/2016/texmf-dist/tex/latex/oberdiek/grfext.sty)
(/usr/local/texlive/2016/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
(/usr/local/texlive/2016/texmf-dist/tex/latex/caption/ltcaption.sty)
*geometry* driver: auto-detecting
*geometry* detected driver: pdftex
*geometry* verbose mode - [ preamble ] result:
* driver: pdftex
* paper: <default>
* layout: <same size as paper>
* layoutoffset:(h,v)=(0.0pt,0.0pt)
* modes: 
* h-part:(L,W,R)=(72.26999pt, 469.75502pt, 72.26999pt)
* v-part:(T,H,B)=(72.26999pt, 650.43001pt, 72.26999pt)
* \paperwidth=614.295pt
* \paperheight=794.96999pt
* \textwidth=469.75502pt
* \textheight=650.43001pt
* \oddsidemargin=0.0pt
* \evensidemargin=0.0pt
* \topmargin=-37.0pt
* \headheight=12.0pt
* \headsep=25.0pt
* \topskip=11.0pt
* \footskip=30.0pt
* \marginparwidth=59.0pt
* \marginparsep=10.0pt
* \columnsep=10.0pt
* \skip\footins=10.0pt plus 4.0pt minus 2.0pt
* \hoffset=0.0pt
* \voffset=0.0pt
* \mag=1000
* \@twocolumnfalse
* \@twosidefalse
* \@mparswitchfalse
* \@reversemarginfalse
* (1in=72.27pt=25.4mm, 1cm=28.453pt)

(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/ucsencs.def)
(/usr/local/texlive/2016/texmf-dist/tex/latex/hyperref/nameref.sty
(/usr/local/texlive/2016/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsfonts/umsa.fd)
(/usr/local/texlive/2016/texmf-dist/tex/latex/amsfonts/umsb.fd)

LaTeX Warning: No \author given.

(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/data/uni-79.def)
(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/data/uninames.dat)
(/usr/local/texlive/2016/texmf-dist/tex/latex/ucs/data/uni-79.def)

! Package ucs Error: Unknown Unicode character 20320 = U+4F60,
(ucs)                possibly declared in uni-79.def.
(ucs)                Type H to see if it is available with options.

See the ucs package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.284 你
         们这些磨人的小妖精啊．非要老夫弄什么中文，弄�...

? 
! Emergency stop.
 ...                                              
                                                  
l.284 你
         们这些磨人的小妖精啊．非要老夫弄什么中文，弄�...

!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on notebook.log.
```

I am running master version of notebook and master version of nbconvert (which is supposed to be using xelatex but based on these logs it's still using pdflatex).

**Comment by michaelpacer** [comment]
@gnestor I'm pretty sure that invoking XeLaTeX can sometimes involve a call to pdfLaTeX if Latexmk is involved…

However, that should definitely not be running pdfTeX…which suggests that something is wrong. I'm going to dive into the nbconvert code to see what's up.

**Comment by michaelpacer** [comment]
Ok…

When you use `nbconvert --to pdf` it errs and seems to be using `pdflatex` not `xelatex` even on the current master. I can't figure out why it's still calling pdflatex. But, that is the source of the error. 

So it looks like when you use the notebook's download as pdf function it invokes everything correctly. It produces a pdf without error. But that doesn't produce any chinese characters in the output (but I think that's a font related issue, more on that later). This should just mirror the nbconvert function, so I have no idea why it wouldn't have the same error.

And, if you convert to LaTeX `nbconvert --to latex` and then manually run xelatex, you get no errors, but the final pdf still doesn't have Chinese characters. I think we don't provide a font with better support for unicode points reserved for CJK values.  I don't know whether one is shipped with the MacTeX package…I just think we might need to declare it explicitly as an option using the `\setCJKmainfont{}` and `\setCJKfallbackfamilyfont{}`…which means that we might need to identify a default CJK font that we make sure people have. 

The problem is you can only declare those things in the preamble which is the part of the LaTeX document that we don't give people access to as far as I can tell in nbconvert. I'll open an issue there, because really this should be an issue there.

**Comment by michaelpacer** [comment]
Ok. there are some fonts included in MacTeX…good so we're set. I'm going to make a PR that adds the relevant commands to the preamble. It may not be the prettiest CJK font, but it's better than having nothing.

Also, once I made these changes, I stopped having issues with --to pdf trying to use pdfTeX and the download as PDF command even worked in the browser. I think this may be because `\usepackage{xeCJK}` is specific to XeLaTeX, so if it were to try to call pdfLaTeX that will break that call and it may fall back to XeLaTeX. This isn't a clean way of solving why sometimes pdfTeX would be used as the actual typesetting engine…but I can't figure out why that was happening at all.
